File: javadevelopers-understanding-kotlin-properties/src/test/java/com/moelholm/jav/ThreeTests.java
package com.moelholm.jav;

import static org.assertj.core.api.Java6Assertions.assertThat;

import com.moelholm.Three;
import org.junit.Test;

public class ThreeTests {

  @Test
  public void test() {

    Three three = new Three();

    three.setMessageOne("Hello from Java");

    assertThat(three.getMessageOne()).isEqualTo("Hello from Java");

    assertThat(three.getMessageTwo()).isNull();

  }

}

File: javadevelopers-understanding-kotlin-properties/src/test/kotlin/com/moelholm/kot/ThreeTests.kt
package com.moelholm.kot

import com.moelholm.Three
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class ThreeTests {

  @Test
  fun test() {

    val three = Three()

    three.messageOne = "Hello from Java"

    assertThat(three.messageOne).isEqualTo("Hello from Java")

    assertThat(three.messageTwo).isNull()

  }

}